src/sync/integrations/github.ts

```
@@ -0,0 +1,446 @@
+						return moment(comment.created_at).unix();
+					});
+					const tailMessages = _.map(
+						sortedComments,
+						Integration.translateMessage
+					);
+					return {
+						messages: _.concat([initialMessage], tailMessages),
+						thread: Integration.translateThread(props.issue.data, [
+							params.repo,
+						]),
+					};
+				});
+			})
+			.catch((error) => {
+				if (retries > 0) {
+					return this.getContext(url, retries - 1);
+				} else {
+					throw new Error(error);
+				}
+			});
+	}
+
+	public push(
+		payload: ChatContext,
+		retries = this.retries
+	): Promise<ChatContext> {
+		const messagesToPush = _.clone(payload.messages);
+		const url = _.get(payload.thread, 'url');
+		const threadPromise = url
+			? Promise.resolve(payload)
+			: this.pushThread(payload.thread, messagesToPush.shift(), 0);
+		return threadPromise
+			.then((threadWrittenContext: ChatContext) => {
+				const writtenThread = threadWrittenContext.thread;
+				const messagePromises = _.map(messagesToPush, (message) => {
+					return this.pushMessage(writtenThread, message, 0).then(
+						(messageWrittenContext) => {
+							return messageWrittenContext.messages;
+						}
+					);
+				});
+				return Promise.all(messagePromises).then((writtenReplies) => {
+					return {
+						thread: writtenThread,
+						messages: _.concat(
+							threadWrittenContext.messages,
+							_.flatten(writtenReplies)
+						),
+					};
+				});
+			})
+			.catch((error) => {
+				if (retries > 0) {
+					return this.push(payload, retries - 1);
+				} else {
+					throw new Error(error);
+				}
+			});
+	}
+
+	public translate({ body }: any): Promise<ChatContext[]> {
+		try {
+			const data = JSON.parse(body);
+			const context: ChatContext = {
+				thread: Integration.translateThread(data.issue, [data.repository.name]),
+				messages: [Integration.translateMessage(data.comment)],
+			};
+			return Promise.resolve([context]);
+		} catch {
```

Is the intention to swallow any potential error in the whole block, or just to deal with `JSON.parse`? We could be swallowing important things here, so probably better to just wrap `JSON.parse` inside the try/catch clause